manifests/init.pp

```diff
@@ -51,25 +53,27 @@
 # @author Ger Apeldoorn <info@gerapeldoorn.nl>
 #
 class rhsm (
-  $rh_user             = undef,
-  $rh_password         = undef,
-  $org                 = undef,
-  $activationkey       = undef,
-  $pool                = undef,
-  $proxy_hostname      = undef,
-  $proxy_port          = undef,
-  $proxy_user          = undef,
-  $proxy_password      = undef,
-  $baseurl             = 'https://cdn.redhat.com',
-  $servername          = 'subscription.rhsm.redhat.com',
-  $serverprefix        = '/subscription',
-  $serverport          = 443,
-  $repo_ca_cert        = '%(ca_cert_dir)sredhat-uep.pem',
-  $manage_repos        = 1,
-  $full_refresh_on_yum = 0,
-  $package_ensure      = 'latest',
-  $repo_extras         = false,
-  $repo_optional       = false
+  $rh_user               = undef,
+  $rh_password           = undef,
+  $org                   = undef,
+  $activationkey         = undef,
+  $pool                  = undef,
+  $proxy_hostname        = undef,
+  $proxy_port            = undef,
+  $proxy_user            = undef,
+  $proxy_password        = undef,
+  $baseurl               = 'https://cdn.redhat.com',
+  $servername            = 'subscription.rhsm.redhat.com',
+  $serverprefix          = '/subscription',
+  $serverport            = 443,
+  $ca_cert_dir           = '/etc/rhsm/ca/',
+  $repo_ca_cert_filename = 'redhat-uep.pem',
+  $repo_ca_cert_source   = undef,
+  $manage_repos          = 1,
+  $full_refresh_on_yum   = 0,
+  $package_ensure        = 'latest',
+  $repo_extras           = false,
+  $repo_optional         = false
```

Comment:
Could you add datatypes at least for the new parameters?

Reply:
I haven't done this by intention and added the datatypes in the Puppet Strings documentation. The reason is our agents are still Puppet 3.8.6 but the catalog is compiled on Puppet 4 masters. Having datatypes currently breaks a few things until we've migrated all agents and tooling.